js_modules/dagit/packages/core/src/hooks/useQueryPersistedState.tsx

```diff
@@ -97,7 +97,7 @@ export function useQueryPersistedState<T extends QueryPersistedDataType>(
       }
 
       currentQueryString = next;
-      history.replace(`${location.pathname}?${qs.stringify(next, {arrayFormat: 'brackets'})}`);
+      history.push(`${location.pathname}?${qs.stringify(next, {arrayFormat: 'brackets'})}`);
```

Comment:
Not sure why this is currently `replace` instead of `push`. I think it would be kind of nice to be able to navigate back to previous filter states via the Back button.

Reply:
Oh hmm I'm not sure about this being the default - I think we have a lot of places where an input field reads/writes the state, so this would yield a history.push on every keystroke. For non-text inputs this seems like a reasonable choice though, I wonder if we could get fancy and stack the changes within a certain timeframe into one history item...

Reply:
Good point. I'll change it back for now and we can investigate further.